src/Data/Nat/Divisibility/Core.agda

```diff
@@ -35,7 +35,30 @@ record _∣_ (m n : ℕ) : Set where
   constructor divides
   field quotient : ℕ
         equality : n ≡ quotient * m
-open _∣_ using (quotient) public
+
+  quotient≡0⇒n≡0 : quotient ≡ 0 → n ≡ 0
+  quotient≡0⇒n≡0 q≡0 rewrite equality | q≡0 = refl
+
+  quotient≢0 : .{{NonZero n}} → NonZero quotient
+  quotient≢0 = ≢-nonZero (contraposition quotient≡0⇒n≡0 (≢-nonZero⁻¹ n))
```

Comment:
This can't be the only place in the library when this kind of encapsulated reasoning about `NonZero`ness based on an implication might be useful?

A subsequent PR might ought to refactor this to `Data.Nat.*` somewhere?

Reply:
See latest commit...